### PR TITLE
Add MAF Complete data type

### PR DESCRIPTION
**Pull request by qu8n**
# HERA Sample Tracker Support: MAF Complete

General updates:
- Defined a new MafComplete class
- Defined Tempo's relationship with MafComplete and provided Tempo class functions to interact with MafComplete
- Defined Neo4j queries to get MafComplete data
- Modified the Tempo message handler to be able to handle MafComplete events 
- Added mocked test data

---
## Crossing T's and dotting I's

Please follow these checklists to help prevent any unexpected issues from being introduced by the changes in this pull request. If an item does not apply then indicate so by surrounding the line item with `~~` to strikethrough the text. See [basic writing and formatting syntax](https://docs.github.com/en/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for more information.

### I. Web service and data model checklist

Please follow these checks if any changes were made to any classes in the web, service, or persistence layers.

**Data checks:**
Updates were made to the mocked incoming request data and/or mocked published request data:
- [x] [smile-server test data](https://github.com/mskcc/smile-server/tree/master/service/src/test/resources/data)
- [ ] [smile-commons test data](https://github.com/mskcc/smile-commons/tree/master/src/test/resources/data)
- [ ] [smile-label-generator test data](https://github.com/mskcc/smile-label-generator/tree/master/src/test/resources/data)
- [ ] [smile-request-filter test data](https://github.com/mskcc/smile-request-filter/tree/master/src/test/resources/data)

**Code checks:**
- [ ] Endpoints were tested to ensure their integrity.
- [ ] Screenshots have been provided to demonstrate changes made to the response body JSON schema and/or swagger page.
- [ ] Unit tests were updated in relation to updates to the mocked test data.

### II. Neo4j models and database schema checklist:
- [x] Neo4j persistence models were changed.
- [ ] The graph database produces the expected changes to models, relationships, and/or property names. [provide screenshot of updated elements in graph db below]

### III. Message handlers checklist:
- [x] Changes in this PR affect the workflow of incoming messages.
- [ ] Messages are following the expected workflow when published to the topic(s) changed or introduced in this pull request.
- [ ] Unit tests were added to ensure messages are handled as expected.

Please describe how the workflow and messaging was tested/simulated:

**Describe your testing environment:**

- NATS [local, local docker, dev server, production]
- Neo4j [local, local docker, dev server, production]
- SMILE Server [local, local docker, dev server, production]
- Message publishing simulation [nats cli, docker nats cli, smile publisher tool, other (describe below)]

Other: [insert details on how messages were published or simulated for testing]

### IV. Configuration and/or permissions checklist:
- [x] New topics were introduced.
- [x] The topics and appropriate permissions were updated in [cmo-metadb-configuration](https://github.mskcc.org/cmo/cmo-metadb-configuration).
- [ ] If applicable, a new account was set up and the account credentials and keys are checked into [cmo-metadb-configuration](https://github.mskcc.org/cmo/cmo-metadb-configuration).
- [ ] Account credentials and keys were shared with the appropriate parties.

---
### Screenshots

---
### General checklist:
- [ ] All requested changes and comments have been resolved.
- [x] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)

### CODE REVIEW

**Comment by qu8n** [comment]
@ao508 Just requesting your review for now. I'll wait to merge it after your review and after rebasing it when your QcComplete work is merged.

Whenever we meet, could we walk through the checkboxes of `smile-server`'s PR description template again? Thank you.

File: model/src/main/java/org/mskcc/smile/model/tempo/MafComplete.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Constructor for MafComplete.
+     * @param date
+     * @param normalPrimaryId
+     * @param status
+     */
+    public MafComplete(String date, String normalPrimaryId, String status) {
+        this.date = date;
+        this.normalPrimaryId = normalPrimaryId;
```
Comment (qu8n) [inline model/src/main/java/org/mskcc/smile/model/tempo/MafComplete.java:30]: Note that while the incoming JSON names this property `normal_primaryId`, I'm renaming it using camel case to fit Java's naming convention.

File: service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java [hunk 2]
```diff
@@ -165,4 +238,29 @@ public void onMessage(Message msg, Object message) {
...
+        gateway.subscribe(TEMPO_WES_MAF_COMPLETE_TOPIC, Object.class, new MessageConsumer() {
+            @Override
+            public void onMessage(Message msg, Object message) {
+                try {
+                    String mafCompleteJson = mapper.readValue(
+                        new String(msg.getData(), StandardCharsets.UTF_8),
+                        String.class);
+                    Map<String, String> mafCompleteMap = mapper.readValue(mafCompleteJson, Map.class);
+                    MafComplete mafComplete = new MafComplete(mafCompleteMap.get("date"),
+                            mafCompleteMap.get("normal_primaryId"),
```
Comment (ao508) [inline service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java:253]: update

**Review by ao508 — COMMENTED** [review]
TODO: already discussed making updates to `normal_primaryId` -> `normalPrimaryId`

File: model/src/main/java/org/mskcc/smile/model/tempo/Tempo.java [hunk 3]
```diff
@@ -130,6 +132,54 @@ public Boolean hasQcCompleteEvent(QcComplete qcComplete) {
         return Boolean.FALSE;
     }
 
+    /**
+         * Returns list of maf complete events.
```
Comment (ao508) [inline model/src/main/java/org/mskcc/smile/model/tempo/Tempo.java:136]: indentation

File: service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java [hunk 4]
```diff
@@ -141,12 +148,53 @@ public void run() {
...
+                        }
+                    }
+                } catch (InterruptedException e) {
+                    interrupted = true;
+                } catch (Exception e) {
+                    LOG.error("Error during handling of MAF complete event", e);
+                }
+            }
+        }
+
```
Comment (ao508) [inline service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java:188]: extra new line

File: service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java [hunk 5]
```diff
@@ -261,4 +333,30 @@ public void onMessage(Message msg, Object message) {
             }
         });
     }
+
```
Comment (ao508) [inline service/src/main/java/org/mskcc/smile/service/impl/TempoMessageHandlingServiceImpl.java:336]: extra newline

**Review by ao508 — CHANGES_REQUESTED** [review]
there are some very small stylistic changes to fix but the code itself is good to go

once updated please feel free to merge the PR yourself

File: model/src/main/java/org/mskcc/smile/model/tempo/Tempo.java (hunk 3 above)
Comment (qu8n) [reply model/src/main/java/org/mskcc/smile/model/tempo/Tempo.java]: Nice catch. Thank you

**Pull request merged**